bundles/catalogue/metadataflyout/service/MetadataLoader.js

```diff
@@ -35,6 +35,10 @@ Oskari.clazz.define(
                 '&lang=' + lang +
                 '&srs=' + this.srs;
 
+            if (metadataUrl !== null) {
+                uri = uri + '&metadataUrl=' + metadataUrl;
+            }
```

Comment:
Sending an URL to be called is problematic from security perspective. We should just send the layer id instead (removing the uuid) and let server get the optional address and the uuid based on the layer id.